### DISCUSSION TITLE
PDF Error

### DISCUSSION THREAD

**Discussion by vrendon852** [code4recovery/12-step-meeting-list#804] [Q&A] [answered]
I'm creating this post to request support help for the AA CSO (Central Service Office) Website here in San Antonio.

- URL: aasanantonio.org
- Description of what I'm experiencing: (all screenshots attached)

1. Log into WP Admin Portal
2. Along left hand side select "Meetings" then "Import and Settings"
![12StepMtg_Plugin_ss1](https://user-images.githubusercontent.com/105313936/167703168-198f639e-c325-4353-9157-1475ab4fd612.jpg)

3. Middle of next page, under "Where's my Info?", 2 paragraphs down where it says: 
"Going away soon: a very basic PDF schedule is available in three sizes: 4×7, half page and full page."
4. Click on "half page" hyperlink
![12StepMtg_Plugin_ss2](https://user-images.githubusercontent.com/105313936/167703172-6a948aa8-afde-4182-8cd5-573ad62ff7a1.jpg)
5. Error Message (SS then Text):
![12StepMtg_Plugin_ss3_error](https://user-images.githubusercontent.com/105313936/167703175-b27bd00d-637f-44ef-88f1-c753085c1391.jpg)
Warning: Undefined array key "" in /var/www/wp-content/plugins/12-step-meeting-list/includes/pdf.php on line 140

Fatal error: Uncaught Exception: TCPDF ERROR: Some data has already been output, can't send PDF file in /var/www/wp-content/plugins/12-step-meeting-list/includes/tcpdf/tcpdf.php:2995 Stack trace: #0 /var/www/wp-content/plugins/12-step-meeting-list/includes/tcpdf/tcpdf.php(7684): TCPDF->Error('Some data has a...') #1 /var/www/wp-content/plugins/12-step-meeting-list/includes/ajax.php(88): TCPDF->Output('meeting-schedul...', 'I') #2 /var/www/wp-includes/class-wp-hook.php(307): tsml_ajax_pdf('') #3 /var/www/wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters('', Array) #4 /var/www/wp-includes/plugin.php(474): WP_Hook->do_action(Array) #5 /var/www/wp-admin/admin-ajax.php(187): do_action('wp_ajax_tsml_pd...') #6 {main} thrown in /var/www/wp-content/plugins/12-step-meeting-list/includes/tcpdf/tcpdf.php on line 2995
There has been a critical error on this website.

Learn more about troubleshooting WordPress.
----------
Additional Notes:
* The CSO Manager is needing to convert our AA SA Meeting Guide into a 'half-page' PDF due to a request to print hard copies for local entities.

* CSO Manager says this worked just 5 days ago (half page and Generate PDF button)

* I believe I updated the PHP on this site 3+ weeks ago (can't find any notes on this at the moment)

- Version Info: AA CSO Website: Version 5.9.3
- Plugin Version Info:

----12 Step Meeting List----
Version: 3.14.2

Last Updated: 2 weeks ago
Requires WordPress Version: 3.2 or higher
Compatible up to: 5.9.3
Requires PHP Version: 5.6 or higher

----12 Step Meeting List - WordPress - Plug-In----
Version: 0.3.1

Last Updated: 7 months ago
Requires WordPress Version: 4.7 or higher
Compatible up to: 5.8.4
Requires PHP Version: 5.6 or higher

----
Any and all suggestions are welcome. I greatly appreciate you taking the time to reply/comment/help.
-Vince R.

**Comment by joshreisner** [comment, marked as answer]
Hi @vrendon852 

Looks like you've discovered a bug in the old PDF generator -- it seems to be struggling with a meeting that is by appointment (it doesn't have a day set). Looks like it's this meeting: https://aasanantonio.org/meetings/open-door-group/ -- I am pretty sure that if you set a day on that meeting, or remove it, the PDF will start working again.

That said, we invite you to check out the new PDF generator, since like it says, this one is going away soon.

**Reply by tech2serve** [reply]
@vrendon852, 

Thank you for taking the time to write up such a good Q&A. That is one of the most complete I've seen, and it really helps us help you!

Tim

**Comment by vrendon852** [comment]
Josh-
Thank you for the technical feedback.
I'll test to see if that resolves our issue.
I appreciate your quick response.
I'll update you on this asap.

Regards,

Vincent Rendon
ClearTech Owner/IT Specialist
***@***.***<https://cleart3ch.com/wp-content/uploads/2020/12/CT_logo_white_on_blue_email_autosig.png>
Cell: 210-730-8489
***@***.******@***.***>
www.cleart3ch.com<http://www.cleart3ch.com/>
SBE,MBE,DBE,TX HUB

Schedule a meeting with me: Clear Tech ***@***.***/bookings/>

CONFIDENTIALITY NOTICE: This message and any accompanying documents contain information belonging to the sender which may be confidential and legally privileged. This information is only for the use of the individual or entity to which it was intended.

**Comment by vrendon852** [comment]
Tim-
You're welcome.

Being in the IT Industry for many years I've experienced how things can get long and drawn out if the due diligence isn't taken on the front end. How else can I expect help if I'm not clear, thorough and concise.

Appreciate the compliment and glad it was helpful in providing me with a timely response.
Thank you for all that you do. Greatly appreciate it.

Regards,

Vincent Rendon
ClearTech Owner/IT Specialist
***@***.***<https://cleart3ch.com/wp-content/uploads/2020/12/CT_logo_white_on_blue_email_autosig.png>
Cell: 210-730-8489
***@***.******@***.***>
www.cleart3ch.com<http://www.cleart3ch.com/>
SBE,MBE,DBE,TX HUB

Schedule a meeting with me: Clear Tech ***@***.***/bookings/>

CONFIDENTIALITY NOTICE: This message and any accompanying documents contain information belonging to the sender which may be confidential and legally privileged. This information is only for the use of the individual or entity to which it was intended.
